spec/middleware/ga4_optimise_middleware_spec.rb

```diff
@@ -0,0 +1,28 @@
+require_relative "../../lib/middleware/ga4_optimise_middleware"
+
+describe Ga4OptimiseMiddleware do
+  subject(:ga4_optimise_middleware) { described_class.new(app) }
+
+  let(:env) { Rack::MockRequest.env_for }
+  let(:app) { ->(_env) { ActionDispatch::Response.create(200, {}, html).to_a } }
+
+  context "when the html contains an unoptimised data-ga4- attribute" do
+    let(:html) { '<a data-ga4-link="my &quot;attribute&quot;">' }
+
+    it "replaces the escaped quotes in a data-ga4- attribute with unescaped ones" do
+      expected_output = "<a data-ga4-link='my \"attribute\"'>"
+
+      (_, _, body) = ga4_optimise_middleware.call(env)
+      expect(body.first).to eq(expected_output)
+    end
+  end
+
+  context "when the html contains an unoptimised non-data-ga4- attribute" do
+    let(:html) { '<a data-attribute="my &quot;attribute&quot;">' }
+
+    it "leaves the input alone" do
+      (_, _, body) = ga4_optimise_middleware.call(env)
+      expect(body.first).to eq(html)
+    end
+  end
+end
```

Comment:
Do we need to test apostrophes as well for the quote checks? E.g. the heading on this page has apostrophes which get added to `data-ga4-form`: https://www.gov.uk/maternity-paternity-calculator/y/paternity/england/no/2026-01-01/2026-01-01 

(I appreciate that the example I've given isn't in `email-alert-frontend` though.)